bindings/csharp/Regorus/Regorus.cs

```diff
@@ -0,0 +1,305 @@
+using System.Runtime.InteropServices;
+using System.Text;
+using System;
+
+#nullable enable
+namespace Regorus
+{
+	public sealed class Exception : System.Exception
+	{
+		public Exception(string? message) : base(message) { }
+	}
+
+	public sealed class Engine : System.IDisposable
```

Comment:
Not sure if there's any downside to just marking the whole class as `unsafe`, to avoid having to wrap individual blocks in `unsafe`.

```csharp
public sealed unsafe class Engine : IDisposable
{
```

Reply:
Nice idea! That simplified the code a whole lot.